test/helpers/getModelsMapForPopulate.test.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const assert = require('assert');
+const start = require('../common');
+const util = require('../util');
+
+const mongoose = start.mongoose;
+const Schema = mongoose.Schema;
+
+describe('getModelsMapForPopulate', function() {
+  let db;
+
+  beforeEach(() => db.deleteModel(/.*/));
+
+  before(function() {
+    db = start();
+  });
+
+  after(async function() {
+    await db.close();
+  });
+
+  afterEach(() => util.clearTestData(db));
+  afterEach(() => util.stopRemainingOps(db));
+
+  it('should error on missing options on populate', async function() {
+    const sch = new Schema({
+      test: mongoose.Schema.Types.ObjectId
+    }, {
+      virtuals: {
+        someVirtual: {}
+      }
+    });
+
+    const model = db.model('Test', sch);
+
+    const doc = await model.create({ test: new mongoose.Types.ObjectId() });
+
+    await assert.rejects(() => model.findById(doc._id).populate('someVirtual').exec(), /If you are populating a virtual, you must set the localField and foreignField options \(Model: Test\)/);
```

Comment:
I think it would be more helpful to also include the virtual path in case the user is populating multiple paths. Maybe something more like `"Mongoose can't populate virtual ${virtual.path} on model ${model} because it doesn't have a localField or foreignField set"` ?

Reply:
is 66c553042def63af1d96c45b9ac5e183b8f6d1e2 better?